api/types/swarm/runtime/plugin.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+import "gogoproto/gogo.proto";
+
+// GenericPluginPayload is used to wrap the PluginSpec with extra options before
+// scheduling on the cluster.
+// Its primary use is to pass along registry auth credentials from the client.
+message GenericPluginPayload {
+	PluginSpec spec = 1 [(gogoproto.nullable) = false];
+	PluginPullOptions PullOptions = 2 [(gogoproto.nullable) = false];
```

Comment:
See discussion above about removing this.

The main reason for having it separate is `PluginSpec` specified in json on the service create endpoint, and it shouldn't really contain the registry auth credentials there.
This is similar to what's being done with container-based services.